Orange/widgets/data/owgroupby.py

```diff
@@ -0,0 +1,490 @@
+            cb.setDisabled(True)
+            cb.stateChanged.connect(partial(self.__aggregation_changed, agg))
+            grid_layout.addWidget(cb, i % itm_col, i // itm_col)
+
+    ############
+    # Callbacks
+
+    def __rows_selected(self) -> None:
+        """Callback for table selection change; update checkboxes"""
+        selected_attrs = self.get_selected_attributes()
+
+        types_ = set(type(attr) for attr in selected_attrs)
+        active_aggregations = [self.aggregations[attr] for attr in selected_attrs]
+        for agg, cb in self.agg_checkboxes.items():
+            cb.setDisabled(len(types_ & AGGREGATIONS[agg].types) == 0)
+
+            activated = [agg in a for a in active_aggregations]
+            with block_signals(cb):
+                # check if aggregation active for all selected attributes,
+                # partially check if active for some else uncheck
+                cb.setCheckState(
+                    Qt.Checked
+                    if activated and all(activated)
+                    else (Qt.PartiallyChecked if any(activated) else Qt.Unchecked)
+                )
+
+    def __gb_changed(self) -> None:
+        """
+        Callback for Group-by attributes selection change; update attribute
+        and call commit
+        """
+        rows = self.gb_attrs_view.selectionModel().selectedRows()
+        values = self.gb_attrs_view.model()[:]
+        self.gb_attrs = [values[row.row()] for row in sorted(rows)]
+        # everything cached in result should be recomputed on gb change
+        self.result = Result()
+        self.commit()
+
+    def __aggregation_changed(self, agg: str) -> None:
+        """
+        Callback for aggregation change; update aggregations dictionary and call
+        commit
+        """
+        selected_attrs = self.get_selected_attributes()
+        for attr in selected_attrs:
+            if self.agg_checkboxes[agg].isChecked() and self.__aggregation_compatible(
+                agg, attr
+            ):
+                self.aggregations[attr].add(agg)
+            else:
+                self.aggregations[attr].discard(agg)
+            self.agg_table_model.update_aggregation(attr)
+        self.commit()
+
+    @Inputs.data
+    def set_data(self, data: Table) -> None:
+        self.closeContext()
+        self.data = data
+
+        # reset states
+        self.cancel()
+        self.result = Result()
+        self.Outputs.data.send(None)
+        self.gb_attrs_model.set_domain(data.domain if data else None)
+        self.gb_attrs = data.domain[:1] if data else []
+        self.aggregations = (
+            {
+                attr: DEFAULT_AGGREGATIONS[type(attr)].copy()
+                for attr in data.domain.variables + data.domain.metas
+            }
+            if data
+            else {}
+        )
+
+        self.openContext(self.data)
+
+        # update selections in widgets and re-plot
+        self.agg_table_model.set_domain(data.domain if data else None)
```

Comment:
Domain is already set above, before opening the context (as it should be). Is there any reason to repeat this?

Reply:
Above we set the domain to `gb_attrs_model` (attribute model for the group by list). Here we set the domain to `agg_table_model` which is used for the table. It is set after the context is open to plot the table with settings opened by the context (aggreagtions). Initializing it before the context is open would require another call to the table to replot after the context is open.